overlord/ifacestate/handlers.go

```diff
@@ -350,6 +350,26 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 
 	connRef := interfaces.ConnRef{PlugRef: plugRef, SlotRef: slotRef}
 
+	var plugSnapst snapstate.SnapState
+	if err := snapstate.Get(st, plugRef.Snap, &plugSnapst); err != nil {
+		if autoConnect {
```

Comment:
we should check that err is state.ErrNoState

Reply:
the change looks wrong,   the check needs to be on

if autoConnect && err == state.ErrNoState